### DISCUSSION TITLE
open new tab on form submit

### DISCUSSION THREAD

**Discussion by faizan0210** [orchidsoftware/platform#2581] [Q&A] [answered]
I wanted to open a preview for news in my laravel orchid panel on submitting a form through post. My problem is when users click on preview button the preview is opening on same page but when we press back btn for final submitting the form, the data gets disappeared. So now i m thinking to open the preview on new tab. Anyone can suggest me a better idea to implement will be highly appreciated, I just wanted to show preview but the data should be available on pressing back button which will show thw form page. Thanks in advance

This is my screen
public function commandBar(): array
    {
        $buttons[] = Button::make(__('Save'))
                ->icon('check')
                ->method('save');

         $buttons[] = Button::make('Preview')->id('ddd')
                   ->icon('check')
                   ->method('preview');
                   
        return $buttons;
    }

I wanted to open a preview for news in my laravel orchid panel on submitting a form through post. My problem is when users click on preview button the preview is opening on same page but when we press back btn for final submitting the form, the data gets disappeared. So now i m thinking to open the preview on new tab. Anyone can suggest me a better idea to implement will be highly appreciated, I just wanted to show preview but the data should be available on pressing back button which will show thw form page. Thanks in advance

This is my screen

` public function commandBar(): array
    {
        $buttons[] = Button::make(__('Save'))
                ->icon('check')
                ->method('save');

         $buttons[] = Button::make('Preview')->id('ddd')
                   ->icon('check')
                   ->method('preview');
                   
        return $buttons;
    }`
and my preview function

 `public function preview(NewsItem $new, Request $request)
    {
       

        $newData = $request->get('new');
      
       
       return redirect()->away(route('platform.news.preview', [
             'newData' => $newData, 
         ]))->with('_blank');  // _blank does not works
       
    }`

Even i tried it using Links, but i m not getting the post form data to pass it with route as follows

Link::make('previewww')
                    ->icon('fa.print-solid')
                    ->target('_blank')
                    ->method('preview');
                    // ->route('platform.news.preview');

I m new in laravel orchid and needs support. thanks in advance

**Comment by faizan0210** [comment]
solved by myself

**Reply by maiconrcf** [reply, marked as answer]
How did it?

**Reply by faizan0210** [reply]
Link::make('Preview')->icon('eye')->href('')->id('preview_review'),
in screen made this btn which submits form data to database.

then inside layout of same page, calling view for js as below
Layout::view('preview_js'),


inside view file calling below ajax func. This function will fetch the values from database by id and returns url which on ajax success open on new tab

$("#preview_review").click(function(e) {

    var form = $("#post-form");
    e.preventDefault();
    $.ajax({
        type: form.attr('method'),
        url: "{{route('review_preview')}}",
        data: form.serialize(),
        success: function(data) {
            window.open(data, '_blank');
        }
    })
})


And in web.php, below code
Route::any('/review_preview', [CasesReviewsController::class, 'preview'])->name("review_preview");
this route in web.php

Hope this helps you
